torchinductor/lowering.py

```diff
@@ -2723,12 +2723,6 @@ def mutate_to(changed, val):
         ).data
         assert isinstance(val, ir.StorageBox)
 
-    if isinstance(changed_data, ir.StorageBox):
```

Comment:
This will slow things down.  Perhaps just add a check if changed_data's name is an input?

Reply:
For a single inplace add_ it speeds things up (by not writing to additional copy), but I'll check a few more complicated graphs, and check for the name

Reply:
Ok I restored the shortcut and fixed the output dtype for inplace ops, which solves a few issues with inplace ops, but doesn't solve input mutation. It looks like copying mutation back to input should be handled here https://github.com/pytorch/torchdynamo/blob/41c44bc1d080d6cf063419a4166732b983b84eef/torchinductor/graph.py#L295 yet for some reason it doesn't work - @jansel any idea what's going on?

Reply:
Added output replacement, that seems to work but I don't know what I'm doing. @jansel if that looks fine I'll add the tests.

Reply:
Seems fine to me, not sure why that didn't work.